src/main/java/frc/robot/utilities/PIDCoefs.java

```diff
@@ -1,98 +1,109 @@
 package frc.robot.utilities;
 
+import com.google.gson.annotations.SerializedName;
+
 /**
  * This class is used to store settings for different PIDs
  */
 public class PIDCoefs {
+    @SerializedName(value = "kP", alternate = {"p", "P", "KP"})
+    private double KP;
+    @SerializedName(value = "kI", alternate = {"i", "I", "KI"})
+    private double KI;
+    @SerializedName(value = "kD", alternate = {"d", "D", "KD"})
+    private double KD;
+    @SerializedName(value = "kF", alternate = {"f", "F", "KF"})
+    private double KF;
+    @SerializedName(value = "tolerance", alternate = {"kT", "t", "T", "KT"})
+    private double tolerance;
+    @SerializedName(value = "deltaTolerance", alternate = {"kDT", "dt", "DT", "KDT"})
+    private double deltaTolerance;
+
+    /**
+     * @param KP             The Proportional coefficient of the PID loop in this
+     *                       command.
+     * @param KI             The Integral coefficient of the PID loop in this
+     *                       command.
+     * @param KD             The Differential coefficient of the PID loop in this
+     *                       command.
+     * @param tolerance      The error tolerance of this command.
+     * @param deltaTolerance The tolerance of the change in error.
+     */
+    public PIDCoefs(double KP, double KI, double KD, double tolerance, double deltaTolerance) {
+        this(KP, KI, KD, 0, tolerance, deltaTolerance);
+    }
+
+    /**
+     * @param KP             The Proportional coefficient of the PID loop in this
+     *                       command.
+     * @param KI             The Integral coefficient of the PID loop in this
+     *                       command.
+     * @param KD             The Differential coefficient of the PID loop in this
+     *                       command.
+     * @param KF             The Feed-Forward coefficient of the PID loop in this
+     *                       command.
+     * @param tolerance      The error tolerance of this command.
+     * @param deltaTolerance The tolerance of the change in error.
+     */
+    public PIDCoefs(double KP, double KI, double KD, double KF, double tolerance, double deltaTolerance) {
+        this.KP = KP;
+        this.KI = KI;
+        this.KD = KD;
+        this.KF = KF;
+        this.tolerance = tolerance;
+        this.deltaTolerance = deltaTolerance;
+
+    }
+
+    public PIDCoefs() {
+    }
```

Comment:
what is this constructor? github co-pilot? remove it.

Reply:
you have to have this in order to allow an empty constructor, if you already defined a constructor that's not empty

Reply:
why is there an empty constructor tho?

Reply:
so we don't have to use the regular one with lots of zeroes